WooCommerce/Classes/ViewModels/Authentication/AccountCreationFormViewModel.swift

```diff
@@ -0,0 +1,111 @@
+import Combine
+import Foundation
+import struct Yosemite.CreateAccountResult
+import struct Yosemite.Credentials
+import enum Yosemite.AccountAction
+import enum Yosemite.AccountCreationAction
+import enum Yosemite.CreateAccountError
+import protocol Yosemite.StoresManager
+import class WordPressShared.EmailFormatValidator
+
+/// View model for `AccountCreationForm` view.
+final class AccountCreationFormViewModel: ObservableObject {
+    /// Email input.
+    @Published var email: String = ""
+    /// An error can come from the WPCOM backend, when the email is invalid or already exists.
+    @Published private(set) var emailErrorMessage: String?
+    /// Local validation on the email field.
+    @Published private(set) var isEmailValid: Bool = false
+
+    /// Password input.
+    @Published var password: String = ""
+    /// An error can come from the WPCOM backend, when the password is too simple.
+    @Published private(set) var passwordErrorMessage: String?
+    /// Local validation on the password field.
+    @Published private(set) var isPasswordValid: Bool = false
+
+    private let stores: StoresManager
+    private var subscriptions: Set<AnyCancellable> = []
+
+    init(stores: StoresManager = ServiceLocator.stores) {
+        self.stores = stores
+
+        $email
+            .debounce(for: .seconds(Constants.fieldDebounceDuration), scheduler: DispatchQueue.main)
+            .sink { [weak self] email in
+                self?.validateEmail(email)
+            }.store(in: &subscriptions)
+
+        $password
+            .debounce(for: .seconds(Constants.fieldDebounceDuration), scheduler: DispatchQueue.main)
+            .sink { [weak self] password in
+                self?.validatePassword(password)
+            }.store(in: &subscriptions)
+    }
+
+    /// Creates a WPCOM account with the email and password.
+    /// - Returns: async result of account creation.
+    @MainActor
+    func createAccount() async -> Result<Void, Error> {
+        await withCheckedContinuation { continuation in
```

Comment:
Should we clear the error messages on the field before calling account creation again, to avoid keeping outdated errors?

Reply:
I cleared the error messages separately after input changes in https://github.com/woocommerce/woocommerce-ios/pull/7906/commits/0e80f8d1227cec595dffe1407174b407f9812ae6